robottelo/api/utils.py

```diff
@@ -437,3 +438,34 @@ def configure_provisioning(org=None, loc=None):
         'host_group': host_group.name,
         'domain': domain.name,
     }
+
+
+def get_role_by_bz(bz_id):
+    """Create and configure custom role entity for the testing of specific bugs
+     This function will read the dictionary of permissions and their associated
+     bugzilla id's from robottelo.constants "PERMISSIONS_WITH_BZ",
+     these permissions will create filter and a single role will be created
+     from all the filters.
+
+     :param bz_id: This is the bugzilla id that is specified in the
+        PERMISSIONS_WITH_BZ list, all the permissions associated with the bz_id
+        will be fetched and filters will be created
+     :return: A single role entity will be created from all the created filters
+     """
+    perms_dict = {}
+    role = entities.Role().create()
+    for res_type in PERMISSIONS_WITH_BZ.keys():
+        for value in PERMISSIONS_WITH_BZ[res_type]:
+            if bz_id in value.get('bz', []):
+                if res_type in perms_dict:
+                    perms_dict[res_type].append(value['name'])
+                else:
+                    perms_dict[res_type] = [value['name']]
+    for resource_type in perms_dict.keys():
+        for value in perms_dict[resource_type]:
```

Comment:
I don't think there's a need in nested loops, and especially iterating through nested loops once to form a list of items to actually work with in another 2 nested loops, that's too overkill. You can combine these blocks into one (without forming temporary dict). And regarding nested loops - i'd rather use `.items()`, `filter` and list comprehension to avoid them.
Take a look on this implementation instead:
```python
def get_role_by_bz(bz_id):
    role = entities.Role().create()
    for res_type, perms in PERMISSIONS_WITH_BZ.items():
        perms_with_bz = filter(lambda x: x.get('bz'), perms)
        if perms_with_bz:
            permissions = [
                entities.Permission(
                    name=perm['name'],
                    resource_type=res_type,
                ).search()[0]
                for perm in perms
            ]
            entities.Filter(permission=permissions, role=role).create()
    return role.read()
```
This snippet above already uses my assumptions in comments below.
@rochacbruno your thoughts?